Examples/src/main/kotlin/com/mapbox/vision/examples/POIActivityKt.kt

```diff
@@ -0,0 +1,262 @@
+                poiBitmapRect.bottom = y
+            }
+            return poiBitmapRect
+        }
+
+        private fun calculatePOILabelAlpha(poiState: POIState): Int {
+            val minDistance = min(DRAW_LABEL_MIN_DISTANCE_METERS - poiState.distanceToVehicle, DISTANCE_FOR_ALPHA_APPEAR_METERS)
+            return ((minDistance / DISTANCE_FOR_ALPHA_APPEAR_METERS.toFloat()) * 255).toInt()
+        }
+
+        private fun drawPOIList(poiDrawDataList: List<POIDrawData>): Bitmap {
+            val bitmap = Bitmap.createBitmap(bitmapCameraFrame)
+            val canvas = Canvas(bitmap)
+            for (p in poiDrawDataList) {
+                with(p) {
+                    paint.alpha = poiBitmapAlpha
+                    canvas.drawBitmap(poiBitmap, null, poiBitmapRect, paint)
+                }
+            }
+            return bitmap
+        }
+    }
+
+    override fun onPermissionsGranted() {
+        startVisionManager()
+    }
+
+    override fun initViews() {
+        setContentView(R.layout.activity_poi)
+        poiList = buildPOIList()
+    }
+
+    override fun onStart() {
+        super.onStart()
+        startVisionManager()
+    }
+
+    override fun onStop() {
+        super.onStop()
+        stopVisionManager()
+    }
+
+    override fun onResume() {
+        super.onResume()
+        vision_view.onResume()
+    }
+
+    override fun onPause() {
+        super.onPause()
+        vision_view.onPause()
+    }
+
+    private fun startVisionManager() {
+        if (allPermissionsGranted() && !visionReplayManagerWasInit) {
+            VisionReplayManager.create(sessionPath)
+            VisionReplayManager.visionEventsListener = visionEventsListener
+            vision_view.setVisionManager(VisionReplayManager)
+            VisionReplayManager.start()
+            visionReplayManagerWasInit = true
+        }
+    }
+
+    private fun stopVisionManager() {
+        if (visionReplayManagerWasInit) {
+            VisionReplayManager.stop()
+            VisionReplayManager.destroy()
+            visionReplayManagerWasInit = false
+        }
+    }
+
+    private fun buildPOIList(): List<POI> {
+
+        val poiHamburgers = POI(
+                27.68255352973938,
+                53.94267477012304,
+                getBitmapFromAssets("ic_hamburger.png"))
+
+        val poiGasStation = POI(
+                27.674764394760132,
+                53.9405971055192,
+                getBitmapFromAssets("ic_gas_station.png"))
+
+        val poiHighWay = POI(
+                27.673187255859375,
+                53.940477115649095,
+                getBitmapFromAssets("ic_highway.png"))
+
+        val poiCarWash = POI(
+                27.675944566726685,
+                53.94105180084251,
+                getBitmapFromAssets("ic_car_wash.png"))
+
+        return arrayListOf(poiHamburgers, poiGasStation, poiHighWay, poiCarWash)
+    }
+
+    private fun getBitmapFromAssets(asset: String): Bitmap {
```

Comment:
I'd prefer crash with runtime exception, it is quite easy for developers realize the excact issue

Reply:
sometimes we can build for test users and it may be hard for them to collect logs. 
They have got runtime crash and that's it

maybe create enum of our icons and get name from object with double check on PR

@yunikkk @kiryldz wdyt?

Reply:
I am the one who likes crash-fast in terms of sample code.
We are not building end application that should run smooth in most scenarios - we need to fail-fast so that developer could easily fix it.
IMHO

Reply:
Agree with fail-fast approach here. 
Since this code is a sample code for developers, not for end users - I don't think we'll face problems with crashes from users.

Reply:
okay
but we should be careful with renaming icons and before release new samples check ALL screens